src/modules/billing/services/supplementary-billing-service/supplementary-processor.js

```diff
@@ -203,7 +203,9 @@ const getPairGroupingKey = transaction => {
  * @return {Array<Object>}
  */
 const filterCancellingTransactions = transactions => {
```

Comment:
As before we can chain this instead of mutating, it won't let me make a suggestion for the whole code but if we wanted we could have something like this:
```js
const filterCancellingTransactions = transactions => {
  return transactions
    .sort((a, b) => moment(a.dateCreated).unix() - moment(b.dateCreated).unix())
    .reduce((acc, transaction) => {
      const index = acc.findIndex(row => row.isCredit === !transaction.isCredit)
      index === -1 ? acc.push(transaction) : acc.splice(index, 1)
      return acc
    }, [])
}
```

Reply:
With those changes made it is showing as a code smell on Sonar Cloud. Ive reverted this file back to how it was , unless there is another way to do it :)